projects/packages/sync/src/replicastore/class-table-checksum-user.php

```diff
@@ -0,0 +1,194 @@
+
+		// Prepare the compound key.
+		$key_fields = array();
+
+		// Prefix the fields with the table name, to avoid clashes in queries with sub-queries (e.g. meta tables).
+		foreach ( $this->key_fields as $field ) {
+			$key_fields[] = $this->table . '.' . $field;
+		}
+
+		$key_fields = implode( ',', $key_fields );
+
+		// Prepare the checksum fields.
+		$checksum_fields = array();
+		// Prefix the fields with the table name, to avoid clashes in queries with sub-queries (e.g. meta tables).
+		foreach ( $this->checksum_fields as $field ) {
+			$checksum_fields[] = $this->table . '.' . $field;
+		}
+		// Apply latin1 conversion if enabled.
+		if ( $this->perform_text_conversion ) {
+			// Convert text fields to allow for encoding discrepancies as WP.com is latin1.
+			foreach ( $this->checksum_text_fields as $field ) {
+				$checksum_fields[] = 'CONVERT(' . $this->table . '.' . $field . ' using latin1 )';
+			}
+		} else {
+			// Conversion disabled, default to table prefixing.
+			foreach ( $this->checksum_text_fields as $field ) {
+				$checksum_fields[] = $this->table . '.' . $field;
+			}
+		}
+
+		$checksum_fields_string = implode( ',', array_merge( $checksum_fields, array( $salt ) ) );
+
+		$additional_fields = '';
+		if ( $granular_result ) {
+			// TODO uniq the fields as sometimes(most) range_index is the key and there's no need to select the same field twice.
+			$additional_fields = "
+				{$this->table}.{$this->range_field} as range_index,
+			    {$key_fields},
+			";
+		}
+
+		$filter_stamenet = $this->build_filter_statement( $range_from, $range_to, $filter_values );
+
+		// usermeta join to limit on user_level.
+		$join_statement = "JOIN {$wpdb->usermeta} as um_table ON um_table.user_id = {$this->table}.ID";
+
+		$query = "
+			SELECT
+				{$additional_fields}
+				SUM(
+					CRC32(
+						CONCAT_WS( '#', {$salt}, {$checksum_fields_string} )
+					)
+				)  AS checksum
+			 FROM
+			    {$this->table}
+				{$join_statement}
+			 WHERE
+				{$filter_stamenet}
+				AND um_table.meta_key = '{$wpdb->prefix}user_level'
+			  	AND um_table.meta_value > 0
+		";
+
+		/**
+		 * We need the GROUP BY only for compound keys.
+		 */
+		if ( $granular_result ) {
+			$query .= "
+				GROUP BY {$key_fields}
+				LIMIT 9999999
+			";
+		}
+
+		return $query;
+	}
+
+	/**
+	 * Obtain the min-max values (edges) of the range.
+	 *
+	 * @param int|null $range_from The start of the range.
+	 * @param int|null $range_to   The end of the range.
+	 * @param int|null $limit      How many values to return.
+	 *
+	 * @return array|object|void
+	 * @throws Exception Throws an exception if validation fails on the internal function calls.
+	 */
+	public function get_range_edges( $range_from = null, $range_to = null, $limit = null ) {
+		global $wpdb;
+
+		$this->validate_fields( array( $this->range_field ) );
+
+		// Performance :: When getting the postmeta range we do not want to filter by the whitelist.
```

Comment:
Since this class is for handling User table checksums, is this comment relevant?